File: 18870.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int n, temp;
int arr[1000004];
int arr2[1000004];

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n;

    for(int i=0; i<n; i++){
        cin >> temp;
        arr[i] = temp;
        arr2[i] = temp;
    }

    sort(arr2, arr2+n);

    int last = unique(arr2, arr2+n) - arr2;

    for(int i = 0; i<n; i++){
        int lower = lower_bound(arr2, arr2+last, arr[i]) - arr2;
        cout << lower << " ";    
    }

    return 0;
}

File: 18870.py
import sys
from bisect import bisect_left

n = sys.stdin.readline()
arr = list(map(int, sys.stdin.readline().split()))
arr2 = set()

for a in arr:
    arr2.add(a)

arr2 = list(arr2)
arr2.sort()

for x in arr:
    print(bisect_left(arr2,x), end=" ")
